tools/container/toolkit/toolkit.go

```diff
@@ -793,3 +825,98 @@ func generateCDISpec(opts *Options, nvidiaCDIHookPath string) error {
 
 	return nil
 }
+
+// getRuntimeBinaryPaths extracts the full paths of the low-level runtime binaries specified in the container runtime config
+func getRuntimeBinaryPaths(hostRoot string) ([]string, error) {
+	var runtimeBinaryPaths []string
+
+	if r, ok := os.LookupEnv("RUNTIME"); ok {
+		runtimeConfigCommand := getRuntimeConfigCommand(hostRoot, r)
+		var err error
+
+		switch r {
+		case "containerd":
+			runtimeBinaryPaths, err = getContainerdRuntimeBinaryPaths(runtimeConfigCommand)
+			if err != nil {
+				log.Warningf("Unable to determine containerd runtime binary paths: %v", err)
+			}
+
+		case "crio":
+			runtimeBinaryPaths, err = getCrioRuntimeBinaryPaths(runtimeConfigCommand)
+			if err != nil {
+				log.Warningf("Unable to determine crio runtime binary paths: %v", err)
+			}
+		}
+	}
+	return runtimeBinaryPaths, nil
+}
+
+// getContainerdRuntimeBinaryPaths extracts the full paths of the low-level runtime binaries specified in the containerd runtime config
+func getContainerdRuntimeBinaryPaths(cliArgs []string) ([]string, error) {
```

Comment:
This should probably be a `engine.Config` method and be implemented in the `engine/containerd` package.

Reply:
I am not sure I agree. The `getContainerdRuntimeBinaryPaths` method invokes the containerd implementation of the `engine/config.Interface` interface. I am not sure if we should extract out the method into the same package as the interface method it invokes